### DISCUSSION TITLE
Turning off AuthnRequest signing for SAML client

### DISCUSSION THREAD

**Discussion by krisleech** [keycloak/keycloak#20927] [Q&A]
I believe Keycloak is expecting each SAML Request (AuthnRequest) to be signed. Our Service Provider does not support signed SAML Requests.

The error I am getting is `SigAlg was null`:

```
request validation failed: org.keycloak.common.VerificationException: SigAlg was null
2023-06-09 10:57:55,436 WARN  [org.keycloak.events] (executor-thread-72) type=LOGIN_ERROR, realmId=d162e9db-f175-47fb-bef4-249f144b453f, clientId=null, userId=null, ipAddress=172.17.0.1, error=invalid_signature
```

In the metadata I see `WantAuthnRequestsSigned="true"`, I think this needs to be false.

I’m using KeyCloak 21.1.1 in Docker.

Any help or pointers would be much appreciated.

**Comment by wind-shift** [comment]
It would be good, as of today I don’t have the answer yet and it’s quite blocking. Other IdP have not enabled this setting so this makes it awkward to integrate keycloak as an IdP at this point, and it conflicts with other settings namely signing SP requests which is actually disabled.

**Reply by embesozzi** [reply]
You can disable the requirement that the authn request not be signed. Keycloak supports that. I've worked with several SAML applications that both did and did not sign their authentication requests.
